tests/run_tests.sh

```diff
@@ -269,7 +269,7 @@ libfdt_overlay_tests () {
     for test in $BAD_FIXUP_TREES; do
 	tree="overlay_bad_fixup_$test"
 	run_dtc_test -I dts -O dtb -o $tree.test.dtb "$SRCDIR/$tree.dts"
-	run_test overlay_bad_fixup overlay_base_no_symbols.test.dtb $tree.test.dtb
+	run_test overlay_bad_fixup overlay_base_manual_symbols.test.dtb $tree.test.dtb
```

Comment:
This change is unrelated, logically or code-wise to the other one.  Can you please make them separate patches / commits (they can still go in the same PR on github).

Reply:
The test case update is necessary due to the refactoring, as it's essential for the tests to pass with the modified code logic. Specifically, the change prevents premature test exits due to the updated phandle handling. For the detailed rationale, please refer to point 2 in my original commit message. Nonetheless, I've split the commits for clarity as you suggested.

Reply:
Sorry, I missed that.  I had the impression that the first patch didn't make any behavioural change, just optimised the existing behaviour.  Since it clearly does, I need to consider more carefully if this is the right idea.

I'm having trouble understanding point 2 from your commit message.  Can you clarify for me why the behaviour is different with the overlay change?

Reply:
To clarify point two, the key difference between `overlay_base_manual_symbols.test.dtb` and `overlay_base_no_symbols.test.dtb` is that the former includes a `__symbols__` node which is crucial for our updated logic.

I'd like to draw attention to this code segment:

```c
symbol_off = fdt_path_offset(fdt, symbol_path);
if (symbol_off < 0)
	return symbol_off;
```
This block's role is to locate the `__symbols__` node in the base fdt. Before refactoring, the code checked the format of fixup string(like "/fragment@0:target:ab") in the fdto before looking up the `__symbols__` in the base fdt. After refactoring, we first check for the `__symbols__` node in the base fdt.

Therefore, post-refactor, the `__symbols__` node must exist in the base fdt to ensure the code evaluates fixup strings correctly and doesn't exit early due to missing `__symbols__`.

I hope this clarification is helpful. Let me know if further details are needed.

Reply:
Apologies for the mix-up. The code block I previously posted was incorrect. Here’s the correct one.
```c
/* And base DTs without symbols */
symbols_off = fdt_path_offset(fdt, "/__symbols__");
if ((symbols_off < 0 && (symbols_off != -FDT_ERR_NOTFOUND)))
	return symbols_off;
```
Also, my earlier explanation was not entirely accurate. Simply put, we need the base fdt to contain a `__symbols__` node so that the test case will process the logic for checking the fixup string format. Without this, the test would prematurely exit with an FDT_ERR_NOTFOUND error at ```symbol_path = fdt_getprop(fdt, symbols_off, label, &prop_len);```

Reply:
Ah, I see.  Previously the generated trees here had two defects: first they were missing the symbols, and second the actual defect we introduced deliberately to test.  With your other change we're now failing on that first defect instead of the second one we're trying to test.